### DISCUSSION TITLE
Why does Renovate parse lockfiles instead of integrating with native tools?

### DISCUSSION THREAD

**Discussion by hkrutzer** [renovatebot/renovate#23614] [Request Help] [answered]
### How are you running Renovate?

Placeholder value, please select the correct response from the dropdown

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Hi,

In the past I've tried to use Renovate to manage an Elixir project.
However, I ran into too many issues similar to https://github.com/renovatebot/renovate/issues/18462, that would not occur if Renovate were to use an Elixir script to retrieve Elixir dependencies, like how Dependabot does it:https://github.com/dependabot/dependabot-core/blob/main/hex/helpers/lib/parse_deps.exs

So I was wondering why Renovate uses Javascript to parse lockfiles/dependency files for all supported languages, instead of using code that runs in the language that it's updating dependencies for, where this might be less error-prone?

### What you tried so far.

N/a

**Comment by viceice** [comment, marked as answer]
we did that in the past and it was even more error prone.

also performance was very bad.
